pinot-hadoop/src/main/java/org/apache/pinot/hadoop/job/SegmentPreprocessingJob.java

```diff
@@ -505,6 +377,86 @@ private void addHashCodeField(Set<Schema.Field> fieldSet) {
     fieldSet.add(hashCodeField);
   }
 
+  private void setTableConfigAndSchema() throws IOException {
+    // If push locations, table config, and schema are not configured, this does not necessarily mean that segments
+    // cannot be created. We should allow the user to go to the next step rather than failing the job.
+    if (_pushLocations.isEmpty()) {
+      _logger.error("Push locations cannot be empty. "
+          + "They are needed to get the table config and schema needed for this step. Skipping pre-processing");
+      throw new RuntimeException("Push locations cannot be empty.");
+    }
+    try(ControllerRestApi controllerRestApi = new DefaultControllerRestApi(_pushLocations, _rawTableName)) {
+      _tableConfig = controllerRestApi.getTableConfig();
+      _pinotTableSchema = controllerRestApi.getSchema();
+    }
+
+    if (_tableConfig == null) {
+      _logger.error("Table config cannot be null. Skipping pre-processing");
+      return;
+    }
+
+    if (_pinotTableSchema == null) {
+      _logger.error("Schema cannot be null. Skipping pre-processing");
+    }
+  }
+
+  private void setValidationConfigs(Job job, Path path) throws IOException {
+    SegmentsValidationAndRetentionConfig validationConfig = _tableConfig.getValidationConfig();
+
+    // TODO: Serialize and deserialize validation config by creating toJson and fromJson
+    // If the use case is an append use case, check that one time unit is contained in one file. If there is more than one,
+    // the job should be disabled, as we should not resize for these use cases. Therefore, setting the time column name
+    // and value
+    if (validationConfig.getSegmentPushType().equalsIgnoreCase("APPEND")) {
+      job.getConfiguration().set(InternalConfigConstants.IS_APPEND, "true");
+      String timeColumnName = _pinotTableSchema.getTimeFieldSpec().getName();
+      job.getConfiguration().set(InternalConfigConstants.TIME_COLUMN_CONFIG, timeColumnName);
+      job.getConfiguration().set(InternalConfigConstants.SEGMENT_TIME_TYPE, validationConfig.getTimeType().toString());
+      job.getConfiguration().set(InternalConfigConstants.SEGMENT_TIME_FORMAT, _pinotTableSchema.getTimeFieldSpec().getOutgoingGranularitySpec().getTimeFormat());
+      job.getConfiguration().set(InternalConfigConstants.SEGMENT_PUSH_FREQUENCY, validationConfig.getSegmentPushFrequency());
+      _dataStreamReader = getAvroReader(path);
```

Comment:
Make this try-with-resource and close the reader after reading the value. The problem here is that, if the use case is not APPEND use case, then `cleanup()` will throw NPE. You can replace `_dataStreamReader` to a local variable and remove `cleanup()` method.